### ISSUE TITLE
No references in `var` declaration

### ISSUE THREAD

**Issue by DmitrySoshnikov** [eslint/eslint#4240]
Version: latest

The `references` property of a `variable` is an empty array in case of the `var` kind of a variable declaration:

```js
var foo = 10; foo;
```

the `references` is `[]`.

```js
let foo = 10; foo;
```

the `references` contains two references (the init, and the actual reference).

Bug or "by design"?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
This is correct. Try again inside an IIFE. Globals don't have static lexical references. Also, this would have been more appropriately opened against estools/escope.

**Comment by nzakas** [comment]
Yeah, this is more of an escope question.

**Comment by DmitrySoshnikov** [comment]
OK, thanks for clarifications, I'll follow up with `escope`.
